src/chttpd/src/chttpd_sup.erl

```diff
@@ -99,6 +102,13 @@ append_if_set({Key, Value}, Opts) when Value > 0 ->
 append_if_set({_Key, 0}, Opts) ->
     Opts;
 append_if_set({Key, Value}, Opts) ->
+    ?LOG_ERROR(#{
+        what => invalid_config_setting,
+        section => chttpd_auth_cache,
+        key => Key,
+        value => Value,
+        details => "value must be a positive integer"
```

Comment:
Message doesn't indicate whither 0 is allowed or not. In erlang specs there are two distinct types http://erlang.org/doc/reference_manual/typespec.html:

- non_neg_integer() - `0..`
- pos_integer() - `1..`
- neg_integer() - `..-1`

Which doesn't make things clearer IMO because the logs are read by users/operators not Erlang developers.

Maybe we should say something like

- `value must be a positive integer [0..`
- `value must be a positive integer 0..`
- `value must be a positive integer including 0`

Reply:
Oh you know what, I think I just overlooked the second function clause there. "Positive integer" means 1.., this should read "value must be a non-negative integer"